src/groupByKey.ts

```diff
@@ -1,7 +1,26 @@
+import { error } from 'console';
+
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | null {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const name = (item[key] as string) || null;
+
+    if (name) {
+      if (result[name]) {
+        continue;
+      }
+      result[name] = items.filter((data) => data[key] === name);
```

Comment:
Using `items.filter` inside the loop for each item is inefficient as it results in a quadratic time complexity. Instead, consider building the groups incrementally by adding each item to the appropriate group in the `result` object.